### DISCUSSION TITLE
How to customize layout size for custom fields

### DISCUSSION THREAD

**Discussion by goxiaoy** [strapi/strapi#6517] [Legacy Topics]
I notice in relative codes in
https://github.com/strapi/strapi/blob/8d0c6848ba421680570e974e5d8606314735de3e/packages/strapi-plugin-content-manager/services/utils/configuration/layouts.js#L13-L34

I want to config the latout size for my custom field, not MAX_ROW_SIZE / 2=6 only, but I do not know how.

If there is not a way currently, I am glad to PR for this

1, extend registerField  api to add extra properties for custom field, so that content-management plugin or any other plugin can read for this extra properties
2, extend content-management  plugin to make size configurable for each field type

**Comment by goxiaoy** [comment]
See #5919

And I find out the source code is in
https://github.com/strapi/strapi/blob/941f14a07483b8a606e296ea8492167b76ba6541/packages/strapi-plugin-content-manager/admin/src/utils/layout.js#L74-L92

Not in the previous code

**Reply by lpanjwani** [reply]
Hi @Goxiaoy,  Could you elaborate on how you changed the size? I tried extending via `./my-app/extensions` and `./my-app/admin` folders, it doesn't work for me.

**Reply by goxiaoy** [reply]
After a search of latest code, you need to override two parts of code:

Guides: https://strapi.io/documentation/developer-docs/latest/concepts/customization.html#plugin-extensions
You will need to replace into `./my-app/extensions/content-manager/` not `./my-app/admin`

1 This part is used in initial data and validation, which will be called every time content-manager bootstrap
https://github.com/strapi/strapi/blob/8b9c3f2052ba449c653a9c4b90bdf09f98cf477d/packages/strapi-plugin-content-manager/services/utils/configuration/layouts.js#L13-L33

2 This part is used in ui
https://github.com/strapi/strapi/blob/be4d5556936cf923aa3e23d5da82a6c60a5a42bc/packages/strapi-plugin-content-manager/admin/src/containers/EditSettingsView/utils/layout.js#L78-L95

**Reply by lpanjwani** [reply]
Yup, Thanks for the detailed guide. However, it does not override the files using `./my-app/extensions/content-manager/`.

Just to confirm the steps:
1) Override `/my-app/extensions/content-manager/services/utils/configuration/layouts.js`
2) Override `/my-app/extensions/content-manager/admin/src/containers/EditSettingsView/utils/layout.js`
3) Delete `build` & `.cache`.
4) Rebuild or `yarn develop --watch-admin`

I tried even changing the default to test changes yet no changes are visible in the `Single Entry View` and `Configure View` aswell
![image](https://user-images.githubusercontent.com/10087186/103139814-97a66e80-46f9-11eb-8cc6-e8abb9afcb96.png)

Would appreciate any help

**Reply by goxiaoy** [reply]
Try to replace step 1 with:
1 Override `/my-app/extensions/content-manager/services/` folder from https://github.com/strapi/strapi/tree/master/packages/strapi-plugin-content-manager/services, do your changes

If that does not work, try to replace step 1 with
1 Override `/my-app/extensions/content-manager` folder  from https://github.com/strapi/strapi/tree/master/packages/strapi-plugin-content-manager , do your changes

In my previous project, I fully customize the plugin with override the whole plugin folder. I have not viewed how this customization work, so it may be a bug of strapi

**Reply by goxiaoy** [reply]
It may be related with #8950

**Reply by lpanjwani** [reply]
I think its a bug since I have done customizations without overriding the whole folder.
However I'll still give a try aswell and get back to you. 

Thanks for the help 👍

**Reply by goxiaoy** [reply]
In https://github.com/strapi/strapi/blob/048e6fd3afe7e96bcf4e75ccff7ce78cd1bc89e9/packages/strapi/lib/core/load-extensions.js#L23
extension folder replace files exclude subdirectories. So the `/my-app/extensions/content-manager/services/utils/configuration/layouts.js` single file will not work, the minimum replacement would be `/my-app/extensions/content-manager/services/` folder. It may be a design action or a bug
